### PR TITLE
Add Spring-specific options to UI

### PR DESCRIPTION
**Pull request by sofurihafe**
## Description

This PR adds Spring-specific options to UI like ability to choose the configuration.
See the following screenshots:
image
image
image

### Manual tests

Open Action Menu in Spring Project.

### CODE REVIEW

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt [hunk 1]
```diff
@@ -156,6 +156,8 @@ private const val SAME_PACKAGE_LABEL = "same as for sources"
 
 private const val WILL_BE_INSTALLED_LABEL = " (will be installed)"
 
+private const val NO_SPRING_CONFIGURATION_OPTION = "No configuration"
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt:159]: Possibly `Select configuration...`?

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt [hunk 2]
```diff
@@ -181,6 +183,14 @@ class GenerateTestsDialogWindow(val model: GenerateTestsModel) : DialogWrapper(m
 
     private val codegenLanguages = createComboBox(CodegenLanguage.values())
     private val testFrameworks = createComboBox(TestFramework.allItems.toTypedArray())
+
+    //
+    private val isSpringProject = true
+    private val modelSpringConfigs = (listOf(NO_SPRING_CONFIGURATION_OPTION) + model.getSortedSpringConfigurationClasses().map { it.name }).toTypedArray()
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt:189]: Let's prepare mapping to array directly in `get...Classes()` method

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt [hunk 3]
```diff
@@ -181,6 +183,14 @@ class GenerateTestsDialogWindow(val model: GenerateTestsModel) : DialogWrapper(m
 
     private val codegenLanguages = createComboBox(CodegenLanguage.values())
     private val testFrameworks = createComboBox(TestFramework.allItems.toTypedArray())
+
+    //
+    private val isSpringProject = true
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt:188]: We can obtain real value here using Spring dependencies analyzer written by Ilya (in other place I see the TODO, may be you missed it here)

Comment (EgorkaKulikov) [reply utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt:188]: Do we really need this duplicate of `model.applicationType`?

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt [hunk 4]
```diff
@@ -518,8 +536,10 @@ class GenerateTestsDialogWindow(val model: GenerateTestsModel) : DialogWrapper(m
         model.testSourceRoot?.apply { model.updateSourceRootHistory(this.toNioPath().toString()) }
 
         //TODO: obtain this values from UI controls https://github.com/UnitTestBot/UTBotJava/issues/1929
-        model.applicationType = ApplicationType.PURE_JVM
-        model.typeReplacementApproach = TypeReplacementApproach.DO_NOT_REPLACE
+        model.applicationType = ApplicationType.PURE_JVM /* wait for Ilya's PR#1956 merge */
+        model.typeReplacementApproach =
+            if (springConfig.item == NO_SPRING_CONFIGURATION_OPTION) TypeReplacementApproach.DO_NOT_REPLACE
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt:541]: Rewrite it with `when`, please

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt [hunk 5]
```diff
@@ -958,6 +978,18 @@ class GenerateTestsDialogWindow(val model: GenerateTestsModel) : DialogWrapper(m
             updateTestFrameworksList(parametrizedTestSource)
         }
 
+        springConfig.addActionListener { _ ->
+            val isSpringConfigSelected = springConfig.item != NO_SPRING_CONFIGURATION_OPTION
+            if (isSpringConfigSelected) {
+                mockStrategies.item = MockStrategyApi.OTHER_CLASSES
+                mockStrategies.isEnabled = false
+                updateMockStrategyListForConfigGuidedTypeReplacements()
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt:986]: Let's leave a comment here that we now have this UI item in mock strategies list, but it actually represents existing strategy combined with type replacements approach.
Here we would like to convenient for user, but it is very unclear for developers, we need to document it carefully.

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt [hunk 6]
```diff
@@ -996,8 +1028,9 @@ class GenerateTestsDialogWindow(val model: GenerateTestsModel) : DialogWrapper(m
                 || currentFrameworkItem == TestNg && findSdkVersion(model.srcModule).feature > minSupportedSdkVersion
         val mockStrategyIsSupported = mockStrategies.item == MockStrategyApi.NO_MOCKS
 
+        // We do not support PUT in Spring projects
         parametrizedTestSources.isEnabled =
-            languageIsSupported && frameworkIsSupported && mockStrategyIsSupported
+            !isSpringProject && languageIsSupported && frameworkIsSupported && mockStrategyIsSupported
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt:1033]: Think about this version:
```java
val projectTypeSupported = model.applicationType != PURE_JVM
parametrizedTestSources.isEnabled = projectTypeSupported && ...
```

By the way, maybe we should rename applicationType to projectType.

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt [hunk 7]
```diff
@@ -1018,6 +1051,27 @@ class GenerateTestsDialogWindow(val model: GenerateTestsModel) : DialogWrapper(m
...
     }
 
+    private fun updateMockStrategyListForConfigGuidedTypeReplacements() {
+        mockStrategies.renderer = object : ColoredListCellRenderer<MockStrategyApi>() {
+            override fun customizeCellRenderer(
+                list: JList<out MockStrategyApi>, value: MockStrategyApi,
+                index: Int, selected: Boolean, hasFocus: Boolean
+            ) {
+                this.append(value.displayName, SimpleTextAttributes.REGULAR_ATTRIBUTES)
+                this.append(" (with config guided type replacements)", SimpleTextAttributes.REGULAR_ATTRIBUTES)
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt:1061]: This text should be discussed with @alisevych

**Review by EgorkaKulikov — CHANGES_REQUESTED** [review]

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt (hunk 7 above)
Comment (alisevych) [reply utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/ui/GenerateTestsDialogWindow.kt:1061]: @sofurihafe @EgorkaKulikov 
Suggestion: 
"Mock using Spring BeanDefinitions"
or 
"Mock using Spring configuration"

It seems to be no good to inform user about mocking outside package/class.
The main thing is that we'll use BeanDefinitions from Spring configuration.

File: utbot-ui-commons/src/main/kotlin/org/utbot/intellij/plugin/models/BaseTestModel.kt [hunk 8]
```diff
@@ -108,7 +107,7 @@ open class BaseTestsModel(
                 .searchPsiClasses(annotation, searchScope)
                 .findAll()
                 .sortedBy { testRootToIndex[it.containingFile.sourceRoot] ?: Int.MAX_VALUE }
-        }
+        }.mapNotNull { it.name }
```
Comment (EgorkaKulikov) [inline utbot-ui-commons/src/main/kotlin/org/utbot/intellij/plugin/models/BaseTestModel.kt:110]: fqn

**Review by EgorkaKulikov — APPROVED** [review]

**Pull request merged**